ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -218,7 +218,7 @@ public TransientQueryMetadata executeQuery(
               statement.getConfigOverrides())
           .executeQuery(statement);
       registerQuery(query);
-      primaryContext.registerQuery(query);
+      primaryContext.registerQuery(Optional.empty(), query);
```

Comment:
The `Optional.empty()` makes reference to the stream/table who owns this query. Transient queries do not have such stream/table, so the empty parameter. However, I found that the stream/table that has no persistent queries running can be dropped even if it has transient queries running. Seems a bug in all versions of ksql 'cause I don't see transient queries registered in the referential integrity entries. I will fix this bug in another github issue to review it separately, and understand the following behavior once implemented:
```
ksql> drop stream s1;
Cannot drop S1.
The following queries read from this source: [_confluent-ksql-default_transient_2601039483758810470_1599154165566].
The following queries write into this source: [].
You need to terminate them before dropping S1.
```
^ What if there are several push queries running? Does it makes sense to show the queryID or add another line for # of transient queries reading?

Reply:
ack

> What if there are several push queries running? Does it makes sense to show the queryID or add another line for # of transient queries reading?

I think your suggestion of adding another line for # of transient queries reading is OK. This will become extra important to handle properly when we have scalable push queries (cc @AlanConfluent)